kore/src/Kore/Logger/SMTCondition.hs

```diff
@@ -0,0 +1,49 @@
+{- |
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+
+-}
+
+module Kore.Logger.SMTCondition
+    ( SMTCondition (..)
+    , logSMTCondition
+    ) where
+
+import Kore.Logger
+    ( Entry (..)
+    , Severity (Info)
+    , MonadLog
+    , logM
+    )
+import Kore.Internal.Predicate
+    ( Predicate
+    )
+import Kore.Internal.TermLike
+import Data.Text.Prettyprint.Doc
+    ( Pretty (..)
+    )
+import Data.Typeable
+import Kore.Unparser
+
+newtype SMTCondition variable =
+    SMTCondition
+        { getCondition :: Predicate variable 
+        }
+
+instance InternalVariable variable => Pretty (SMTCondition variable) where
+    pretty SMTCondition { getCondition } =
+        pretty $ unparseToString getCondition
+
+instance
+    (InternalVariable variable, Typeable variable)
+    => Entry (SMTCondition variable)
+  where
+    entrySeverity _ = Info
+
+logSMTCondition
+    :: MonadLog log
+    => InternalVariable variable
+    => Typeable variable
+    => Predicate variable -> log ()
+logSMTCondition =
```

Comment:
As above, please name this `infoEvaluateCondition`. Here we can allow the `variable` type to be a type variable, but we must externalize the variables; see [here](https://github.com/kframework/kore/blob/master/kore/src/Kore/Internal/TermLike/TermLike.hs#L287) for an example. You will probably need to define a helper in `Kore.Internal.Predicate` that wraps `Kore.Internal.TermLike.externalizeFreshVariables`.